### PR TITLE
[TCL30-64] Buttons Into Footer (Subtask)

### PR DESCRIPTION
**Pull request by jssckbl**
## Description

As a subtask of [TCL30-56](https://the-collab-lab.atlassian.net/browse/TCL30-56), this PR incorporates the final design of the functioning buttons into the footer. It also provides basic alignment of the buttons.

The following buttons are updated:
- Map
- List
- Fly to me
- Share my location

## Related Issue

closes TCL30-64, which is a subtask of TCL30-56

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|    | :sparkles: New feature     |
|  ✓   | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

 iPhone 6/7/8 - BEFORE   |   iPhone 6/7/8 - AFTER
:----------------------------------:|:-------------------------------------:
TCL30-64before | Screen Shot 2021-09-02 at 9 17 56 PM

## Testing Steps / QA Criteria

1. `jw-footer-buttons`
2. `npm start`
3. Check that what you see is the same as the "After" image in this PR. 
4. Test the following buttons, that they function as expected:
- Map (`localhost:3000/`)
- List (`localhost:3000/list`)
- Share my location (should switch to screen with purple dot on map where you are currently located)
- Fly to me (when sharing your location, drag the screen so your current location icon cannot be seen on the map. Click 'Fly to me', represented with the right pointing arrow, and you should be brought back to your current location icon)

### CODE REVIEW

File: src/components/FlyToMeButton.js [hunk 1]
```diff
@@ -19,9 +21,13 @@ const FlyToMeButton = ({ disabled = false }) => {
   };
 
   return (
-    <button onClick={handleFlyToMe} disabled={disabled}>
-      Fly to me
-    </button>
+    <Button
+      className="btn btn__icon"
+      label="Fly to me"
+      icon={FlyToMeIcon}
```
Comment (alxmcr) [inline src/components/FlyToMeButton.js:27]: I love that it is so simple. You only need to pass an icon. ❤️

File: src/App.css [hunk 2]
```diff
@@ -174,23 +171,21 @@ nav {
...
-  font-weight: 700;
-  line-height: 1.5;
+  line-height: 1.8;
   cursor: pointer;
   text-decoration: none;
 }
 
 .btn {
-  font-family: var(--font-family-body-text);
+  font-family: var(--font-family-headings);
```
Comment (alxmcr) [inline src/App.css:147]: I feel it's much better with this font-family. Thanks!

File: src/components/Nav.js [hunk 3]
```diff
@@ -1,21 +1,27 @@
...
       </li>
       <li>
-        <NavLink to="/list">List</NavLink>
-      </li>
-      <li>
-        <NavLink to="/settings">Settings</NavLink>
-      </li>
-      <li>
-        <NavLink to="/help">Help</NavLink>
+        <Button
```
Comment (alxmcr) [inline src/components/Nav.js:18]: Awesome! Great change 👍 !

File: src/components/ShareMyLocationButton.js [hunk 4]
```diff
@@ -54,11 +52,12 @@ const ShareMyLocationButton = () => {
   };
 
   return (
-    <button
+    <Button
+      className="btn btn__icon"
+      label="Share location"
+      icon={ShareLocationIcon}
```
Comment (alxmcr) [inline src/components/ShareMyLocationButton.js:58]: 🔥

File: src/components/Nav.js (hunk 3 above)
Comment (jssckbl) [reply src/components/Nav.js:18]: It feels weird to do the link -> button thing, but after reading a few accessibility articles written in the last year, I didn't feel as bad, as long as it is labeled well!

File: src/App.css [hunk 5]
```diff
@@ -59,7 +59,6 @@ picture {
   display: block;
 }
 
-button,
```
Comment (alxmcr) [inline src/App.css:62]: @jssckbl  I don't know if you can see, but when you move the mouse over `map` or `list` button, the icon change color. I couldn't see it again. 
image

File: src/App.css [hunk 6]
```diff
@@ -144,13 +143,11 @@ main > h1 + * {
   height: 100%;
 }
 
-footer {
+footer .btn-container {
```
Comment (alxmcr) [inline src/App.css]: @jssckbl  Is it possible?, add a little space between the first  & second row. 
image

Comment (alxmcr) [reply src/App.css]: Possible solution 

image

of course without backgroud-color 😄

File: src/App.css (hunk 5 above)
Comment (jssckbl) [reply src/App.css:62]: I noticed that as well. Still trying to figure it out 😬 

If the user is clicking between `Map` and `List` then the icons remain. 

Once a user clicks on another button other than one of those two, then the icon gets weird again.

File: src/App.css (hunk 6 above)
Comment (jssckbl) [reply src/App.css]: For sure!

**Comment by alxmcr** [comment]
It's not blocking, but it's a small detail. Alignment between buttons.

image

File: src/App.css (hunk 6 above)
Comment (jssckbl) [reply src/App.css]: Added 0.5rem.

File: src/App.css (hunk 5 above)
Comment (jssckbl) [reply src/App.css:62]: I'm not sure how to resolve it yet. But working on it 👍

Comment (jssckbl) [reply src/App.css:62]: Thank you for your help in resolving this, @alxmcr 🥳

File: src/App.css [hunk 7]
```diff
@@ -675,6 +625,10 @@ a.active {
   fill: var(--primary-color);
 }
 
+a.active .icon {
```
Comment (alxmcr) [inline src/App.css:628]: This line is fixing the hover issue 😄 !

File: src/App.css [hunk 8]
```diff
@@ -684,6 +638,55 @@ a.active {
   gap: 1rem;
 }
 
+/* Footer */
```
Comment (alxmcr) [inline src/App.css:641]: Organize a little code ❤️ !

**Review by alxmcr — APPROVED** [review]
Good job, congrats! Thank you so much 🥳 !

File: src/App.css (hunk 7 above)
Comment (jssckbl) [reply src/App.css:628]: Nice! I was suspecting a combo of some of those terms, but it likely would have taken me much longer than it took you to accomplish this. Thank you so much!

**Pull request merged**